src/Reflection/ReflectionProperty.php

```diff
@@ -448,6 +450,8 @@ private function assertClassExist(string $className) : void
      * @throws NoObjectProvided
      * @throws NotAnObject
      * @throws ObjectNotInstanceOfClass
+     * 
+     * @psalm-assert object $object
```

Comment:
I think this is a good addition, but it adds an error on psalm because it complains that when this method is called, the variable can only be an object and this is redundant.

I think we should suppress the error and keep the assert.

Reply:
I think the assertion could be refined further to `@psalm-assert T`, where `T` is the type for this `ReflectionProperty`'s owner, but that's for another time/discussion.

Indeed, psalm assertions are annoying when put in test scope, because they remove obvious/dumb scenarios :D